typing/env.mli

```diff
@@ -127,8 +127,11 @@ val lookup_cltype:
   ?loc:Location.t -> ?mark:bool ->
   Longident.t -> t -> Path.t * class_type_declaration
 
-val copy_types: string list -> t -> t
-  (* Used only in Typecore.duplicate_ident_types. *)
+type copy_of_types
+val make_copy_of_types: string list -> t -> copy_of_types
+val do_copy_types: copy_of_types -> t -> t
+(** [do_copy_types copy env] will raise [Invalid_argument] if the values in
```

Comment:
The current version will raise a fatal error rather than an `Invalid_argument`, isn'it?

Reply:
Well spotted, updating now.